packages/pwa/app/utils/utils.js

```diff
@@ -183,12 +182,28 @@ export const getConfig = (path) => {
  * Otherwise, it will look for the url config based on the current host
  * If none is found, it will return the default url config at the top level of the config file
  */
-export const getUrlConfig = () => {
-    const {hostname} = new URL(getAppOrigin())
-    const urlConfig = getUrlConfigByHostname(hostname)
-    if (urlConfig) {
-        return urlConfig
+export const getUrlConfig = (path) => {
```

Comment:
I think this might be the wrong point in the code to suggest that partners modify. The reason for this is because we are changing the way a function works so it's output would then effect another function output. Additionally I say this because this function has a clear definition (although the block comment on the top needs some work) or at least is did. Simply put, `getUrlConfig` will get the url settings. That is easy to document and understand. Even if we were to change how the configuration was, e.g. if we made it so that we could override the default url settings on a per site or host bases, that code too would be easy to describe.

John and Myself asked for an interface on how developers might re-implement things to fit their needs, the two points in the project that are "interfaceable" are site resolution and url creation. The interface definition might look something like this if we are strictly talking about the methods we want to be customizable👇 

```
/**
* This function, when given a url string will return the locale and site
* configurations. You can choose to use our default implementation
* or provide your own to suit your needs.
* 
* Example:
* resolveConfigFromUrl('/eu/product?locale=it') =>
*/
const resolveConfigFromUrl(url: string) : ConfigValue {
    // *** default implementation ***
}

/**
* This function, given a base path and configuration object
* will return a fully configured url string correctly formatted
*
* Example:
* const configValue = {
*    site: 'refarch',
*    url: {
*        site: 'path', 
*        locale: 'query_param'
*    },
*    locale: {
*        alias: 'en'
*    }
* }
* createUrlWithConfig('/products/1', configValue)
* => '/refarch/product/1?locale=en'
*/
const createUrlWithConfig(path: string, config: ConfigValue) : string {
    // *** default implementation ***
}
```

Note in the above example `ConfigValue` object is the interesting thing to solve, what exactly is this type, it's like the config object but only has information for the current site. Almost like an "evaluated" configuration object.